quinn-proto/src/connection.rs

```diff
@@ -522,35 +519,46 @@ where
     /// Executes protocol logic, potentially preparing signals (including application `Event`s,
     /// `EndpointEvent`s and outgoing datagrams) that should be extracted through the relevant
     /// methods.
-    pub fn handle_timeout(&mut self, now: Instant, timer: Timer) {
-        match timer.0 {
-            TimerKind::Close => {
-                self.state = State::Drained;
-                self.endpoint_events.push_back(EndpointEventInner::Drained);
-            }
-            TimerKind::Idle => {
-                self.close_common();
-                self.events.push_back(ConnectionError::TimedOut.into());
-                self.state = State::Drained;
-                self.endpoint_events.push_back(EndpointEventInner::Drained);
-            }
-            TimerKind::KeepAlive => {
-                trace!("sending keep-alive");
-                self.ping();
-            }
-            TimerKind::LossDetection => {
-                self.on_loss_detection_timeout(now);
-            }
-            TimerKind::KeyDiscard => {
-                self.zero_rtt_crypto = None;
-                self.prev_crypto = None;
-            }
-            TimerKind::PathValidation => {
-                debug!("path validation failed");
-                self.path_challenge = None;
-                self.path_challenge_pending = false;
-                if let Some(prev) = self.prev_path.take() {
-                    self.path = prev;
+    pub fn handle_timeout(&mut self, now: Instant) {
+        for &timer in &Timer::VALUES {
+            match self.timers[timer] {
```

Comment:
Style nit: maybe an `if let .. else` would be a more readable way of expressing this?

Reply:
The double match also feels suboptimal, but I can't think of a better way right now.

Reply:
Reprised.